salt/pillar/consul_pillar.py

```diff
@@ -263,6 +301,24 @@ def pillar_format(ret, keys, value):
     return dict_merge(ret, pil)
 
 
+def nest_tree(pillar, pillar_root):
```

Comment:
I sort of wonder if it would be a little cleaner just to move these inside of the `ext_pillar` function or to a utils dir since they aren't used anywhere but there so far as I can see.

Reply:
That makes sense, I've moved it into ext_pillar and added a couple unit tests as suggested by @isbm